core/extractors/extractors_robust.py

```
@@ -0,0 +1,260 @@
+                    child.terminate()
+                    child.join(timeout=2)
+                    if child.is_alive():
+                        logger.warning(f"Force killing child process: {child.pid}")
+                        child.kill()
+                        child.join(timeout=1)
+                
+                # Additional cleanup of executor processes
+                try:
+                    # Try public shutdown API first
+                    executor.shutdown(wait=False, cancel_futures=True)
+                except Exception:
```

Ruff (0.14.13)

Do not catch blind exception: `Exception`

(BLE001)